src/assets/DockerCompose/docker_compose.md

```diff
@@ -0,0 +1,211 @@
+  2. create a docker compose yml file `dockercompose/compose.yml` inside it.
+  3. Also create an empty `testdump.sql` file (will mention the use of it later)
+
+  It should look something like this.
+  ```
+  📦dockercompose
+      ┗ 📜compose.yml
+      ┗ 📜testdump.sql
+   ```
+
+### Defining and configuring services
+
+Now inside the `compose.yml` lets define and configure the services for our application.
+
+  1. database service (mysql)
+
+      ```yml
+     version: "3.8"
+
+     services:
+        database:
+          image: mysql
+          environment:
+            MYSQL_ROOT_PASSWORD: root
+            MYSQL_DATABASE: demodb
+          volumes:
+            - ./testdump.sql:/docker-entrypoint-initdb.d/testdump.sql
+          restart: always
+     ```
+
+      Here, defining services in docker compose we use keyword `services`. We have defined our service `database` configured with `image:mysql` and `environment` variables required for it.
+      We can see it has a volumes keyword used here. Simply through volume we can mount our local data to the docker container. In this context we have mapped `testdump.sql` which is in our
+      local root directory and ran as an entrypoint when the container starts. It means when `database` service is started in a container the `testdump.sql` is the first thing which run inside the container.
+      And entrypoint script basically creates 4 users in the database. So we need to add some sql command inside of `testdump.sql` as:
+      Like this we defined and configured database service for our application.
+
+      ```sql
+        CREATE TABLE users(userid INT PRIMARY KEY AUTO_INCREMENT, fullname VARCHAR(60), email VARCHAR(60));
+        INSERT INTO users (fullname, email) VALUES ('Alice Smith', 'alice@example.com');
+        INSERT INTO users (fullname, email) VALUES ('Bob Johnson', 'bob@example.com');
+        INSERT INTO users (fullname, email) VALUES ('Eva White', 'eva@example.com');
+        INSERT INTO users (fullname, email) VALUES ('Michael Brown', 'michael@example.com');
+        ALTER USER 'root' IDENTIFIED WITH mysql_native_password BY 'root';
+        flush privileges;
+     ```
+  2. backend service
+
+      ```yml
+        backend:
+          image: sagar4321/demo-server
+          environment:
+            PORT: 3001
+            DBHOST: database
+            DBPORT: 3306
+            DBUSER: root
+            DBNAME: demodb
+            DBPASSWORD: root
+          restart: always
+          ports:
+            - "3001:3001"
+          depends_on:
+            - database
+      ```
+
+      Similarly, we define and configure our backend as above with necessary `image:sagar4321/demo-server` (this image is built and pushed to docker hub which I created), `environment` and `port 3001` (port where the server listen)
+      Here, we have another keyword `depends_on` which has value `database` service, that means that `backend` service is only ran after `database` successfully run.
```

Comment:
This is true, but could be misunderstood. `depends_on` without a health check only makes sure that docker started the service, it does NOT check if the database is up and running. If you want to do that, you need to add [healthchecks](https://docs.docker.com/engine/reference/builder/#healthcheck) to the database service

Reply:
i would like to make it very simple for this blog. I have updated the description for `depends_on` for now.